### ISSUE TITLE
Failing tests that I shouldn't be failing?

### ISSUE THREAD

**Issue by larryhynes** [vstakhov/rspamd#617]
I'm testing `rspamd` (1.1.2) , in a simple way by either calling `rspamc` on messages or piping messages from `procmail` to `rspamc`.

My own emails are consistently hitting the `BROKEN_HEADERS`, `R_SPF_FAIL` and `DMARC_POLICY_SOFTFAIL` rules.

I have a valid spf record in place, along with dkim signing, and my mails pass every other public spf, dkim, dmarc test that I've tried on them, so I'm curious as to why they're 'failing' under `spamc`. (I'm assuming the `DMARC_POLICY_SOFTFAIL` is because I don't have a strict dkim policy?)

I can't seem to figure out what constitutes a 'broken header', the only hint that I could find was 'Headers structure is likely broken', in `rules/misc.lua`. I'm sending from `mutt`, via `msmtp` and using OpenBSD's `opensmtpd` for delivery and as far as I know they honour any relevant RFCs.

If anyone could shed any light on this I'd be grateful, thanks.

**Comment by fatalbanana** [comment]
For proper operation rspamd requires some metadata about the message (sender IP & RDNS, SMTP from/rcpt, HELO, ...) to be passed along with the message. Generally this requirement would be satisfied by rmilter or the MTA itself.

You are seeing `R_SPF_FAIL` because no sender IP is being passed. `DMARC_POLICY_SOFTFAIL` suggests your DMARC policy is `none`. Absence of DKIM-related symbols(?) would suggest rspamd is seeing unsigned mail.

I can't comment on `BROKEN_HEADERS`.

**Comment by larryhynes** [comment]
That's great insight, thanks. I'll keep chipping away at it here!

I can't seem to find an `rmilter` package for OpenBSD, and the `rspamd` package doesn't seem to ship with `rmilter` so I'm restricted, for now, to `rspamd` and `rspamc`.

Thanks again.

**Comment by larryhynes** [comment]
It looks to me, from a small sample, that `rspamd` sees `BROKEN_HEADERS` in pretty much every message that I've passed to it. There doesn't seem to be a score, nor any sign of the `BROKEN_HEADERS` symbol in metrics even, so I'm assuming it's no big deal.

**Comment by vstakhov** [comment]
Do you have any samples of messages that trigger this rule?

**Comment by larryhynes** [comment]
This one, and every other message from the OpenBSD mailing list:

```
Return-Path: owner-misc+M157555=larry=larryhynes.com@openbsd.org
Delivered-To: larry@larryhynes.com
Received: from openbsd.org (lists.openbsd.org [192.43.244.163])
	by mail.larryhynes.com (OpenSMTPD) with ESMTPS id da72067e
	TLS version=TLSv1.2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256 verify=NO
	for <larry@larryhynes.com>;
	Tue, 3 May 2016 10:00:49 +0100 (IST)
Received: from openbsd.org (localhost [127.0.0.1])
	by openbsd.org (OpenSMTPD) with ESMTP id a04d8641
	for <larry@larryhynes.com>;
	Tue, 3 May 2016 02:58:41 -0600 (MDT)
Received: from wolfman.devio.us (wolfman.devio.us [98.142.108.75])
	by openbsd.org (OpenSMTPD) with ESMTPS id 5f7b09c8 TLS version=TLSv1.2 cipher=ECDHE-RSA-CHACHA20-POLY1305 bits=256 verify=NO
	for <misc@openbsd.org>; Tue, 3 May 2016 02:58:00 -0600 (MDT)
Received: from localhost (localhost [127.0.0.1])
	by wolfman.devio.us (Postfix) with ESMTP id 6B140201DEB; Tue,  3 May 2016 04:57:59 -0400 (EDT)
X-Virus-Scanned: amavisd-new at devio.us
Received: from wolfman.devio.us ([127.0.0.1])
	by localhost (wolfman.devio.us [127.0.0.1]) (amavisd-new, port 10024) with ESMTP id GYw_mxQjlczQ; Tue,  3 May 2016 04:57:54 -0400 (EDT)
Received: by wolfman.devio.us (Postfix, from userid 2589) id 90148201DDA; Tue,  3 May 2016 04:57:54 -0400 (EDT)
Date: Tue, 3 May 2016 04:57:54 -0400
From: Jiri B <jirib@devio.us>
To: bytevolcano@safe-mail.net
Cc: misc@openbsd.org
Subject: Re: mfs vs tmpfs: advantages and disadvantages
Message-ID: <20160503085754.GA16282@wolfman.devio.us>
References: <20160503170806.00003217@safe-mail.net>
MIME-Version: 1.0
Content-Type: text/plain; charset=us-ascii
In-Reply-To: <20160503170806.00003217@safe-mail.net>
User-Agent: Mutt/1.5.23 (2014-03-12)
List-Help: <mailto:majordomo@openbsd.org?body=help>
List-ID: <misc.openbsd.org>
List-Owner: <mailto:owner-misc@openbsd.org>
List-Post: <mailto:misc@openbsd.org>
List-Subscribe: <mailto:majordomo@openbsd.org?body=sub%20misc>
List-Unsubscribe: <mailto:majordomo@openbsd.org?body=unsub%20misc>
X-Loop: misc@openbsd.org
Precedence: list
Sender: owner-misc@openbsd.org
X-Spam-Scanner: rspamc 1.1.2
X-Spam-Scan-Time: 3.965
X-Spam-Action: no action
X-Spam-Score: -3.79 / 6.00
X-Spam-Level: 
X-Spam-Symbols: MAILLIST,BAYES_HAM,FORGED_SENDER_MAILLIST,R_SPF_ALLOW,
	RCVD_IN_DNSWL_HI,BROKEN_HEADERS,RWL_MAILSPIKE_GOOD,MIME_GOOD
X-TUID: 0Rpi3NvnsNCT

On Tue, May 03, 2016 at 05:08:06PM +1000, bytevolcano@safe-mail.net wrote:
> With tmpfs being in the tree for the last 2+ years (since OpenBSD 5.5),
> I would like to ask, besides the "-P" option in mount_mfs, what is the
> advantage of using mfs over tmpfs?

tmpfs on Bitrig does support snapshots, which can be used as an alternative
to mpf's "-P". If you are interested check it out.
```

This spam email from last night:

```
Return-Path: upyacidkiab@qbep.com.tw
Delivered-To: hello@larryhynes.com
Received: from SKY-20150201SFT.com (223.240.77.185 [223.240.77.185])
	by mail.larryhynes.com (OpenSMTPD) with SMTP id f6046d2c
	for <hello@larryhynes.com>;
	Tue, 3 May 2016 00:32:09 +0100 (IST)
Received: from dmqbdmqbfq (unknown [48.53.201.3])
	by SKY-20150201SFT.com (Postfix) with ESMTP id 79DB65164EE
	for <hello@larryhynes.com>; Mon, 2 May 2016 23:31:54 -0000
Date: Tue, 3 May 2016 07:31:53 +0800
From: "Icy" <upyacidkiab@qbep.com.tw>
To: "hello" <hello@larryhynes.com>
Reply-To: <huixinsoft60@foxmail.com>
Subject: =?GB2312?B?UmFja2luZyBJbnF1aXJ5?=
X-Mailer: Foxmail 5.0 [cn]
Mime-Version: 1.0
Content-Type: text/plain;
	charset="GB2312"
Content-Transfer-Encoding: base64
Content-Disposition: inline
Message-Id: <20160502233154.79DB65164EE@SKY-20150201SFT.com>
X-Spam-Scanner: rspamc 1.1.2
X-Spam-Scan-Time: 0.812
X-Spam: yes
X-Spam-Action: reject
X-Spam-Score: 10.90 / 6.00
X-Spam-Level: ***********
X-Spam-Symbols: INVALID_POSTFIX_RECEIVED,MIME_GOOD,TO_HELLO,BROKEN_HEADERS,
	BAYES_SPAM
X-TUID: lOvWTd5dX77+

R29vZCBtb3JuaW5nLg0KDQpUaGlzIERhbmllbCBmcm9tIEd1YW5nZG9uZyBTbWFjbyBBdXRvbWF0
aW9uIFRlY2hub2xvZ3kgQ28uLEx0ZC4NCg0KU21hY28gaXMgYSByYWNraW5nIHN5c3RlbSBtYW51
ZmFjdHVyZXIgd2l0aCAxNCB5ZWFycycgZXhwZXJpZW5jZaOsIHNwZWNpYWxpemVkIGluIHdhcmVo
b3VzZSByYWNraW5nIHN5c3RlbXMgbGlrZSBzZWxlY3RpdmUgcmFja2luZywgbG9uZyBzcGFuIHJh
Y2tpbmcsIGRyaXYtaW4gcmFja2luZyBtZXp6YW5pbmUgZXRjLiBPRU0gc2VydmljZSBhdmFpbGFi
bGUuDQoNClBsZWFzZSBmZWVsIGZyZWUgdG8gY29udGFjdCBtZSBmb3IgbW9yZSBpbmZvIGlmIHlv
dSBhcmUgaW50ZXJlc3RlZCBpbiB3YXJlaG91c2UgcmFja2luZyBzeXN0ZW1zLg0KDQpUaGFua3Mg
Zm9yIHlvdXIgcHJlY2lvdXMgdGltZS4NCg0KQmVzdCBSZWdhcmRzDQoNCkljeQ0KDQpHdWFuZ2Rv
bmcgU21hY28gQXV0b21hdGlvbiBUZWNobm9sb2d5IENvLixMdGQNClNoYW5nZGkgSW5kdXN0cmlh
bCBEaXN0cmljdCwgTGlhb2J1IFRvd24sIERvbmdndWFuIDUyMzQwMCwgUC5SLiBDaGluYQ0KV2VD
aGF0o7o1OTgyNDA4MDYNClRlbDogKzg2LTc2OS04OTA4ODg4MiAgRXh0LiA4NTkNCkNlbDogKzg2
IDEzNDIzMjkxMzI3IA0KICAgICAgICs4NiAxMzkyOTQyMDExMg0KRmF4OiArODYtNzY5LTg5MTgx
MTk5DQoNCg==
```

Crontab output:

```
Return-Path: root@didot.larryhynes.com
Delivered-To: root@didot.larryhynes.com
Received: from localhost (didot.larryhynes.com [local])
	by didot.larryhynes.com (OpenSMTPD) with ESMTPA id 2005363a
	for <root@didot.larryhynes.com>;
	Tue, 3 May 2016 10:00:01 +0100 (IST)
Date: Tue, 3 May 2016 10:00:01 +0100 (IST)
From: root@didot.larryhynes.com (Cron Daemon)
To: root@didot.larryhynes.com
Subject: Cron <root@didot> /usr/local/bin/relaydb -f /home/larry/.relaydb -lb | pfctl -T replace -t spamd -f -
Auto-Submitted: auto-generated
X-Cron-Env: <SHELL=/bin/sh>
X-Cron-Env: <PATH=/bin:/sbin:/usr/bin:/usr/sbin>
X-Cron-Env: <HOME=/var/log>
X-Cron-Env: <LOGNAME=root>
X-Cron-Env: <USER=root>
Message-Id: <7ca08bf271471426@didot.larryhynes.com>
X-Spam-Scanner: rspamc 1.1.2
X-Spam-Scan-Time: 0.522
X-Spam-Action: no action
X-Spam-Score: -1.89 / 6.00
X-Spam-Level: 
X-Spam-Symbols: BAYES_HAM,MIME_GOOD,DMARC_POLICY_SOFTFAIL,BROKEN_HEADERS,
	ONCE_RECEIVED
X-TUID: ElUr2HIolvcN

no changes.
```

**Comment by vstakhov** [comment]
Well, I'd recommend you to update to 1.2.6 first of all. Perhaps, this issue has been fixed in the stable branch.

**Comment by larryhynes** [comment]
Most recent OpenBSD package is 1.1.2 (installed), and most recent OpenBSD port is 1.1.4.

I'm attempting a build from source (on OpenBSD 5.9) but `cmake` fails as it can't find `libutil.h`. I think libutil is a Linux-only sort of thing and I'm not sure how easy, or possible, it's going to be to get the libutil header files onto my BSD system. :(

**Comment by vstakhov** [comment]
It is actually BSD only header and I don't think it is a critical issue.

**Comment by larryhynes** [comment]
Any hints on how I can get `cmake` to ignore the following? The build stops dead.

```
-- Configuring for BSD system
CMake Error at CMakeLists.txt:409 (MESSAGE):
  Cannot find header libutil.h for package LIBUTIL
```

**Comment by larryhynes** [comment]
I just commented the 'look for libutil.h' part of `CMakeLists.txt`, will see where that leads!

**Comment by larryhynes** [comment]
Hmmm. I can get 1.3.0 to build, but `rspamadm` segfaults. I can't get 1.2.6 to build. I'm going to revert to 1.1.2 and wait until the OpenBSD package gets updated.

**Comment by vstakhov** [comment]
I have added fixes to build rspamd on OpenBSD to both 1.3 and 1.2 branches.
